lineoperations/README

```diff
@@ -0,0 +1,226 @@
+.. contents::
+
+About
+=====
+
+Line Operations is an assortment of simple line functions that can be applied to an open file.
+
+Features
+========
+
+* Remove Duplicate Lines, sorted
+* Remove Duplicate Lines, ordered
+* Remove Unique Lines
+* Remove Empty Lines
+* Remove Whitespace Lines
+* Sort Lines Ascending
+* Sort Lines Descending
+
+Usage
+=====
+
+After the plugins has been installed successfully, load the plugin via Geany's plugin manager and a new menu item in the Tools menu will appear. Click on each menu item to apply the operation on whole  file. See descriptions below to see operations for each menu item.
+
+Notes
+-----
+
+1. Line Operations will **not** make changes to a file until you save the file.
+
+2. New line characters are considered to be part of the line. If the last line does not contain new line characters it will be considered different from a line with new line character(s).
+
+  Example::
+
+    line 1
+
+  is not the same as::
+
+    line 1\r\n
+
+
+Operation Details
+=================
+
+Remove Duplicate Lines
+----------------------
+
+The first occurrence of each duplicate line will remain in the file. The **Sorted** option will sort the file and remove duplicate lines [fast]. The **Ordered** option will keep the same order of lines [slow].
+
+  Example: Suppose a file has the following lines. (#comments added for clarity)
+
+  ::
+
+    Line 2
+    Line 1
+    Line 2    #removed
+    Line 3
+    Line 1    #removed
+    Line 2    #removed
+
+  The **Remove Duplicate Lines, sorted** will change the file into this:
+
+  ::
+
+    Line 1
+    Line 2
+    Line 3
+
+  The **Remove Duplicate Lines, ordered** will change the file into this:
+
+  ::
+
+    Line 2
+    Line 1
+    Line 3
+
+
+
+Remove Unique Lines
+-------------------
+
+Removes all lines that appear only once.
+
+  Example: Suppose a file has the following lines. (#comments added for clarity)
+
+  ::
+
+    Line 2
+    Line 1
+    Line 2
+    Line 3    #removed
+    Line 1
+    Line 2
+
+  The **Remove Unique Lines** will change the file into this:
+
+  ::
+
+    Line 2
+    Line 1
+    Line 2
+    Line 1
+    Line 2
+
+Remove Empty Lines
+------------------
+
+Removes all lines that end with a newline character(s) AND do not have any other characters.
+
+  Example: Suppose a file has the following lines. (#comments, and \n newline characters added for clarity)
+
+  ::
+
+    Line 2\n
+    Line 1\n
+    \n          #removed
+         \n     #NOT removed (contains spaces)
+        Line 1\n
+    Line 2\n
+
+  The **Remove Unique Lines** will change the file into this:
+
+  ::
+
+    Line 2\n
+    Line 1\n
+         \n
+        Line 1\n
+    Line 2\n
+
+
+Remove Whitespace Lines
+-----------------------
+
+Removes all lines that have only white space characters.
+
+A **whitespace character** is one of:
+
+1. `' '` : space
+2. `'\t'`: horizontal tab
+3. `'\f'`: form feed
+4. `'\v'`: vertical tab
+5. `'\r'`: cartridge return
+6. `'\n'`: newline character
```

Comment:
no, I removed this (I don't know what characters Geany/Scintilla considers whitespace with `sci_get_line_indent_position()`).